module/VuFind/src/VuFind/RecordDriver/SolrMarc.php

```diff
@@ -105,7 +124,34 @@ public function getAllSubjectHeadings()
                     }
                     // If we found at least one chunk, add a heading to our result:
                     if (!empty($current)) {
-                        $retval[] = $current;
+                        if ($extended) {
+                            $sources = [
```

Comment:
It probably makes sense to make this a class property as well, for easier customization/extension.